cpp/include/cudf/types.hpp

```diff
@@ -234,6 +234,12 @@ enum class type_id : int32_t {
   NUM_TYPE_IDS  ///< Total number of type ids
 };
 
+/// @brief Indicates whether a function is null-aware or not.
+enum class null_aware : uint8_t {
+  NO  = 0,  ///< The function is not null-aware
+  YES = 1   ///< The function is null-aware
+};
```

Comment:
Optional: More verbose in place of YES/NO?

```suggestion
enum class null_awareness : bool {
  AWARE  = 0,  ///< The function is not null-aware
  UNAWARE = 1   ///< The function is null-aware
};
```

Reply:
Yes, i think so